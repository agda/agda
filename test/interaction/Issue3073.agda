{-# OPTIONS --type-in-type #-}

module _ where

Set' : ∀ i → Set _
Set' i = Set i

postulate
  A : Set
  a : A
  T : Set → Set
  comp : (P : A → Set) (g : (a : A) → P a) → P a
  foo : ∀ i (Q : A → Set' i) → T (comp _ Q)
  -- (no issue if Set' is replaced by Set)

bar : T {!_!} -- Try to give _
bar = foo _ _

-- WAS:
-- A !=< Agda.Primitive.Level of type Set
-- when checking that the expression _ has type Set
-- SHOULD: succeed (with unsolved meta)
