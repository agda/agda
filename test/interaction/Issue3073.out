(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : Set _Q_17 : A → Set' Agda.Primitive.lzero [ at Issue3073.agda:17,13-14 ] " nil)
((last . 1) . (agda2-goals-action '(0)))
(agda2-give-action 0 "_")
(agda2-status-action "")
(agda2-info-action "*All Goals*" "_Q_17 : A → Set' Agda.Primitive.lzero [ at Issue3073.agda:17,13-14 ] " nil)
((last . 1) . (agda2-goals-action '()))
